docs/error-codes.md

```diff
@@ -266,4 +472,154 @@ the error code. For example:
+      RuntimeHelpers.RunClassConstructor(type.TypeHandle);
+  }
+  ```
+
+#### `IL2055` Trim analysis: Call to `System.Reflection.MethodInfo.MakeGenericMethod` can not be statically analyzed. It's not possible to guarantee the availability of requirements of the generic method.
+
+- ILLink currently doesn't analyze `MethodInfo` values and thus can't statically determine the generic method the `MakeGenericMethod` operates on. If the actual method has generic parameters with `DynamicallyAccessedMembersAttribute` ILLink would be required to fulfill the requirements declared by those attributes, but since the ILLink doesn't know the method, it can't determine if such requirements exist.  
+
+  ``` C#
+  void TestMethod()
+  {
+      // IL2055 Trim analysis: Call to `System.Reflection.MethodInfo.MakeGenericMethod` can not be statically analyzed. It's not possible to guarantee the availability of requirements of the generic method.
+      typeof(MyType).GetMethod("MyMethod").MakeGenericMethod(new Type[] { typeof(MyType) });
+  }
+  ```
+
+#### `IL2056` Trim analysis: The assembly 'assembly name' can not be found
+
+- Calling `CreateInstance` with assembly name 'assembly name' which can't be resolved.  
+
+  ``` C#
+  void TestMethod()
+  {
+      // IL2056 Trim analysis: The assembly 'NonExistentAssembly' can not be found
+      Activator.CreateInstance("NonExistentAssembly", "MyType");
+  }
+  ```
+
+#### `IL2058` Trim analysis: Unrecognized value passed to the parameter 'typeName' of method 'CreateInstance'. It's not possible to guarantee the availability of the target type.
+
+- The value passed as the type name to the `CreateInstance` method can't be statically analyzed, ILLink can't make sure that the type is available.  
+
+  ``` C#
+  void TestMethod(string typeName)
+  {
+      // IL2058 Trim analysis: Unrecognized value passed to the parameter 'typeName' of method 'CreateInstance'. It's not possible to guarantee the availability of the target type.
+      Activator.CreateInstance("MyAssembly", typeName);
+  }
+  ```
+
+#### `IL2059` Trim analysis: Unrecognized value passed to the parameter 'assemblyName' of method 'CreateInstance'. It's not possible to guarantee the availability of the target type.
+
+- The value passed as the assembly name or path to the `CreateInstance` method can't be statically analyzed, ILLink can't make sure that the type is available.  
+
+  ``` C#
+  void TestMethod(string assemblyName)
+  {
+      // IL2059 Trim analysis: Unrecognized value passed to the parameter 'assemblyName' of method 'CreateInstance'. It's not possible to guarantee the availability of the target type.
+      Activator.CreateInstance(assemblyName, "MyType");
+  }
+  ```
+
+#### `IL2060` Trim analysis: Unrecognized type value passed to <target description>. It's not possible to guarantee that the requirements declared by the `DynamicallyAccessedMembersAttribute` are met.
```

Comment:
I don't understand this warning message.:-(

Reply:
As per the example, it is used when a value which we can't statically analyze is assigned to something (target) which has data flow annotation - and thus linker can't guarantee that the annotation's requirement is fulfilled. I welcome any suggestions to improve the message.

Reply:
I think this could be something like

```
#### `IL2059` Argument instance used for parameter 'x' in method 'm' doesn't need to meet DynamicallyAccessedMembersAttribute requirements.
```

Reply:
What about:
```
Value passed to <target description> can not be statically determined and may not meet 'DynamicallyAccessedMembersAttribute' requirements.
```

So specifically for a method parameter:
```
Value passed to the parameter 'x' of method 'm' can not be statically determined and may not meet 'DynamicallyAccessedMembersAttribute' requirements.
```

Reply:
I think that's better but `Value passed to the parameter 'x' of method 'm'` should have its own warning code because if someone will be searching for `Value passed to the parameter` it probably won't find this reference easily

Reply:
There are currently 5 cases:
* `parameter 'name' of method 'name'`
* `return value of method 'name'`
* `field 'name'`
* `implicit 'this' parameter of method 'name'`
* `generic parameter 'name' from 'type/method'`

You think we should have separate code for each of them?

Reply:
I'd say yes as they are about construction and would allow us to have better warning message for each case

Reply:
What about 2047 - it uses the same pattern. For example:
```
IL2047: 'DynamicallyAccessedMemberTypes' in 'DynamicallyAccessedMembersAttribute' on parameter 'type' on method 'Derived.TestMethod' don't match overridden parameter 'type' on method 'Base.TestMethod'. All overridden members must have the same 'DynamicallyAccessedMembersAttribute' usage.
```

Reply:
Yeah, that will need the same treatment